### PR TITLE
#184651062 reset password via email

### PR DESCRIPTION
**Pull request by mirindi19**
##What does this PR do?
 - send an email containing the password reset link
 - reset password

##Description of Task to be completed?
 -Reset password via email

##How should this be manually tested?
 -http://localhost:{PORT}/auth/forgot-password
 -http://localhost:{PORT}/auth/reset-password/{token}

##What are the relevant pivotal tracker/Trello stories?
#184651062

### CODE REVIEW

File: src/middleware/checkUser.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import User from '../database/models/user.model.js';
+const checkUser = async (req, res, next) => {
+  const { email } = req.body;
+  const findUser = await User.findOne({
+    where: { email },
+  });
```
Comment (Edcreation) [inline src/middleware/checkUser.js:6]: There is a utils file for this

File: src/utils/token.js [hunk 2]
```diff
@@ -4,9 +4,24 @@ function generateToken(payload) {
   const token = jwt.sign(payload, process.env.JWT_SECRET);
   return token;
 }
+function generateForgetPasswordToken(payload) {
+  const token = jwt.sign(payload, process.env.RESET_PASSWORD_KEY, {
+    expiresIn: '15m',
+  });
+  return token;
+}
```
Comment (Edcreation) [inline src/utils/token.js:12]: LGTM

**Review by Edcreation — CHANGES_REQUESTED** [review]

File: src/routes/user.route.js
Comment (AdrineUWERA) [inline src/routes/user.route.js:1]: can you validate the new password the use will be setting.

**Review by AdrineUWERA — CHANGES_REQUESTED** [review]

**Review by AdrineUWERA — APPROVED** [review]

**Review by Pacific911 — CHANGES_REQUESTED** [review]
check out why the coveralls are failling

**Review by AdrineUWERA — APPROVED** [review]

**Review by Edcreation — APPROVED** [review]

**Review by JosephNgabo — APPROVED** [review]

**Review by Pacific911 — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**